lib/modules/manager/kotlin-script/extract.ts

```diff
@@ -0,0 +1,35 @@
+import { regEx } from '../../../util/regex';
+import { MavenDatasource } from '../../datasource/maven';
+import type { PackageDependency, PackageFile } from '../types';
+
+const dependsOnRegex = regEx(
+  /@file\s*:\s*DependsOn\s*\(\s*(?<replaceString>"(?<groupId>.*):(?<artifactId>.*):(?<version>.*)")\s*\)/g
+);
+const repositoryRegex = regEx(
+  /@file\s*:\s*Repository\s*\(\s*"(?<repositoryName>.*)"\s*\)/g
+);
+
+export function extractPackageFile(fileContent: string): PackageFile | null {
+  const registryUrls: string[] = [...fileContent.matchAll(repositoryRegex)]
+    .map((match) => {
+      return match.groups?.repositoryName;
+    })
+    .filter((x): x is string => x !== null);
+
+  const deps: PackageDependency[] = [
+    ...fileContent.matchAll(dependsOnRegex),
+  ].map((match) => {
+    const dep: PackageDependency = {
+      currentValue: match.groups?.version,
+      depName: `${match.groups?.groupId}:${match.groups?.artifactId}`,
+      replaceString: match.groups?.replaceString,
+      datasource: MavenDatasource.id,
+      registryUrls: registryUrls.length > 0 ? registryUrls : null,
```

Comment:
the lines you are matching the `DependsOn` do they have Fixed version or a range version?

if they have Fixed version should we add rangeStrategy: 'replace' ? since we are just replacing always?

usually the rangeStrategy is chosing in the Versioning and according to gradle 
`export const supportedRangeStrategies: RangeStrategy[] = ['pin'];`

so do we have ranges here? or is it always a fixed version of the dependency?

Reply:
For now I wanted to support fixed versions for simplicity, but I'll check if ranges are supported in this context.

Reply:
After a little digging, Kotlin Script under the hood uses https://github.com/apache/maven-resolver which means that Maven version format is used: https://maven.apache.org/pom.html#Dependency_Version_Requirement_Specification. I tested it quickly and it does work with a Kotlin script. Maybe then `versioning/maven` should be used - I see that the module implements the syntax mentioned in the Maven docs.

Reply:
Great now make lots and lots of tests with all formats and see if it works as expected xD

Reply:
I thought about sanity-testing most obvious cases manually, and then since we delegate dealing with versions to Maven versioning module, we can assume it's already pretty mature right? :smile:

Reply:
Yes, maven versioning and datasource can be assumed to be mature